migration/migration.go

```diff
@@ -1138,4 +1139,167 @@ func removeInvalidUserPreferences(usersPrefs []v1schema.UserPreferences) (update
 	return
 }
 
+func insertGroups(ctx context.Context, mongodb *MongoDB, postgres *PostgresDB) (err error) {
+	dbName := "platform-db"
+	collectionName := "Groups"
+	reqLogger := logf.FromContext(ctx).WithValues("MongoDB.DB", dbName, "MongoDB.Collection", collectionName)
+	filter := bson.M{
+		"$and": bson.A{
+			bson.M{"migrated": bson.M{"$ne": true}},
+			bson.M{"type": bson.M{"$regex": "//^SAML$//i"}},
```

Comment:
So one of two suggestions here; either:
- Just do a direct string comparison - I don't know that the case sensitivity handling should be important, and regexes can be annoying to deal with
- If we need the regex, then this needs to be fixed as this will not work. It would be easier to guarantee that this does what you want with the following form:
```suggestion
			bson.M{"type": bson.M{"$regex": "^SAML$", "$options": "i"}},
```
Alternatively, I think there may be a type in the primitives package that is used for defining regexes.